### ISSUE TITLE
How to implements Keep me signed in functionality  in iOS Swift?

### ISSUE THREAD

**Issue by reddy4** [Azure-Samples/active-directory-b2c-ios-swift-native-msal#64]

**Comment by jasoncoolmax** [comment]
Let me know if this helps:
https://docs.microsoft.com/en-us/azure/active-directory-b2c/session-behavior?pivots=b2c-user-flow#enable-keep-me-signed-in-kmsi

**Comment by reddy4** [comment]
Thanks ::) jason 
I went through above link, it is all about how to configured in Azure and how to develop custom policies.
My requirement is how to implements in iOS Swift Programming Language.

**Comment by antrix1989** [comment]
We don't provide API to turn on KMSI programmaticaly.

**Comment by reddy4** [comment]
Thanks:)) antrix

If not providing API to turn on/of KMSI programmatically? then why allowing us to display option **Keep me signed in** on UI with check box? 
Please refer attached screen shot:
Please clarify below points if possible:
1. We should configure KMSI in Azure portal itself as per microsoft document, but my question is at client side (iOS Mobile app) how to check weather KMSI enabled or NOT?
2. Once ADB2C login successfully do we have any property to identify KMSI Enable or not?


My Requirement is Example:
 I open app at very first time  &   after successfully authenticated with ADB2C navigate inside my application and work around 5 mins, then close my application.  when i open my application second time (after 5-6 hours),  first I need to check KMSI enabled or not based on that i will navigate, If KMSI enable directly i should navigate to dashboard screen , if KMSI not enable then i should navigate to ADB2C Login screen.
 
 Please provide  any suggestion / guidance much appreciate :::))))) 

<img width="753" alt="Screen Shot 2022-08-23 at 3 01 23 PM" src="https://user-images.githubusercontent.com/3033985/186124861-857f0e94-9dc3-41d7-82d9-8029ee49a426.png">

**Comment by reddy4** [comment]
Any help from any one Thanks!

**Comment by reddy4** [comment]
@antrix1989  Any help for above issue?

**Comment by mipetriu** [comment]
Hi @reddy4, I apologize for the silence here. In the example you mentioned above, when you launch your native app you should check if MSAL can acquire a token silently. This will check if a cached token is currently available - KMSI has no effect on this.

If a silent token acquisition is not available, then you could acquire a token interactively. This will open a browser webview with your login UI - normally your user would be prompted for credentials here, but they may not be if KMSI is enabled. This is mentioned in the screenshot of the doc here: "Users who select KMSI, can reopen the browser without being prompted to reenter their username and password."

In other words, you could use the result of the call to MSAL's acquireTokenSilent API to see if a login is needed: https://azuread.github.io/microsoft-authentication-library-for-objc/Classes/MSALPublicClientApplication.html#/c:objc(cs)MSALPublicClientApplication(im)acquireTokenSilentWithParameters:completionBlock:

Please let me know if that would work, if you'd like me to elaborate on any of those points.

**Comment by mipetriu** [comment]
Hi @reddy4, I will close this issue due to inactivity, but please feel free to re-open if you have any more questions. Thank you!
